subsys/bluetooth/host/gatt.c

```diff
@@ -885,7 +886,28 @@ static void sc_indicate(uint16_t start, uint16_t end);
 static void db_hash_process(struct k_work *work)
 {
 #if defined(CONFIG_BT_SETTINGS)
-	if (atomic_test_and_clear_bit(gatt_sc.flags, DB_HASH_LOAD)) {
+	bool hash_loaded_from_settings =
+		atomic_test_bit(gatt_sc.flags, DB_HASH_LOAD);
+	bool already_processed =
+		atomic_test_bit(gatt_sc.flags, DB_HASH_LOAD_PROC);
+
+	if (!hash_loaded_from_settings) {
+		/* we want to generate the hash, but not overwrite the hash
+		 * stored in settings, that we haven't yet loaded.
+		 */
+		db_hash_gen(false);
```

Comment:
can we get `db_hash_store` out of `db_hash_gen()`?

Reply:
done